lib/msf/core/auxiliary/manageengine_xnode/action.rb

```diff
@@ -0,0 +1,65 @@
+# -*- coding: binary -*-
+
+module Msf::Auxiliary::ManageengineXnode::Action
+  # Returns an Xnode authentication request hash
+  #
+  # @param user [String] Username
+  # @param pass [String] Password
+  # @return [Hash] Xnode authentication request 
+  def action_authenticate(user,pass)
+    {
+      "username" => user,
+      "password" => pass,
+      "action" => "session:/authenticate"
+    }
+  end
+
+  # Returns an Xnode health status request hash
+  #
+  # @return [Hash] Xnode health status request 
+  def action_admin_health
+    {
+      "action" => "admin:/health",
+      "de_health" => true,
+      "request_id" => 1,  
+    }
+  end
+
+  # Returns an Xnode info request hash
+  #
+  # @return [Hash] Xnode info request 
+  def action_xnode_info
+    {
+      "action" => "admin:/xnode_info",
+      "request_id" => 1,
+    }
+  end
+
+  # Returns an Xnode DataRespository Search request hash
+  #
+  # @param repo_name [String] Name of the DataRespository to query
+  # @param fields [Array] data respository fields (columns) to dump
+  # @param custom_query [Hash] A hash containing a custom query to merge with the query hash
+  # @return [Hash] Xnode data respository search request
+  def action_dr_search(repo_name, fields=nil, custom_query=nil)
+    time_gen_from = rand(500000000..1000000000) # generate random Unix timestamp somewhere between 1985 - 2001, used below as the "from" date, to ensure we get all data we want
```

Comment:
@gwillcox-r7 no, we could expand the range even further or narrow it down a bit, but the point is that it has to be a date far enough in the past that it would always capture the earliest records that were added to ManageEngine. I didn't want to hardcode one specific date so I figured to go with a safe range and use ` rand`  on it so that the module always picks a different date, but always one far enough in the past. Let me know if you would prefer a different approach

Reply:
Nope this is fine I just wanted to double check the reasoning on this and it sounds fine. Marking this as resolved since your logic sounds good.